doc/user/evpn.rst

```diff
@@ -0,0 +1,379 @@
+RIB (locally originated, learned from a peer, or leaked from another VRF) will
+be eligible to be exported to EVPN so long as they are valid and selected in
+the VRF's unicast table.
+
+In this example, the BGP instances for vrf1 and vrf2 will have their static
+routes redistributed into the BGP loc-rib for the ipv4 unicast and ipv6 unicast
+address-families via the "redistribute static" statements. These unicast
+prefixes will then be exported into EVPN as Type-5 routes as a result of the
+"advertise ipv4 unicast" and "advertise ipv6 unicast" commands.
+
+.. code-block:: frr
+
+   router bgp 4200000000 vrf vrf1
+    !
+    address-family ipv4 unicast
+     redistribute static
+    exit-address-family
+    !
+    address-family ipv6 unicast
+     redistribute static
+    exit-address-family
+    !
+    address-family l2vpn evpn
+     advertise ipv4 unicast
+     advertise ipv6 unicast
+    exit-address-family
+   exit
+   !
+   router bgp 4200000000 vrf vrf2
+    !
+    address-family ipv4 unicast
+     redistribute static
+    exit-address-family
+    !
+    address-family ipv6 unicast
+     redistribute static
+    exit-address-family
+    !
+    address-family l2vpn evpn
+     advertise ipv4 unicast
+     advertise ipv6 unicast
+    exit-address-family
+   exit
+
+.. _evpn-linux-interface-configuration:
+
+Linux Interface Configuration
+=============================
+The Linux kernel offers several options for configuring netdevices for an
+EVPN-VXLAN environment. The following section will include samples of a few
+netdev configurations that are compatible with FRR which implement the
+environment described above.
+
+Some high-level config considerations:
+
+* An L3VNI should always have an SVI (aka the L3-SVI).
+* The L3-SVI should not be assigned an IP address, link-local or otherwise.
+* An SVI for an L2VNI is only needed for routing (IRB) or ARP/ND suppression.
+* MAC/VTEP learning should always be disabled on VXLAN netdevs used for EVPN.
+* The local VTEP-IP should always be set to a reachable IP on the lo device.
+
+.. _evpn-traditional-bridge-traditional-vxlan-devices:
+
+Traditional Bridges and Traditional VXLAN Devices
+-------------------------------------------------
+In the traditional bridge model, we use a separate bridge interface per MAC-VRF
+which acts as the SVI for that broadcast domain.
+
+.. code-block:: shell
+
+   ###################
+   ## vxlan vtep-ip ##
+   ###################
+   ip addr add 100.64.0.1/32 dev lo
+
+   #############################
+   ## ip-vrf vrf1 / l3vni 100 ##
+   #############################
+   ip link add vrf1 type vrf table 1100
+   ip link set vrf1 up
+   ip link add br100 type bridge
+   ip link set br100 master vrf1 addrgenmode none
+   ip link set br100 addr aa:bb:cc:00:00:64
+   ip link add vni100 type vxlan local 100.64.0.1 dstport 4789 id 100 nolearning
+   ip link set vni100 master br100 addrgenmode none
+   ip link set vni100 type bridge_slave neigh_suppress on learning off
```

Comment:
Is `learning off` necessary when `nolearning` is already set on the VXLAN interface?

Reply:
Yes. `type bridge_slave .. learning off` is disabling dynamic MAC learning in the bridge driver whereas `type vxlan .. nolearning` disables dynamic VTEP learning in the vxlan driver. They're quite similar in behavior but are controlled separately and should both be disabled in practice.